### PR TITLE
Release 3.7 Prometheus + Grafana cherry

### PR DESCRIPTION
**Pull request by mjudeikis**
cherry-pick Prometheus to 3.7
cherry-pick Grafana community version to 3.7 release.

### CODE REVIEW

**Comment by mjudeikis** [comment]
@zgalor @mrsiano @sdodson 
Cherry Prometheus and grafana to 3.7.
Grafana is community version, and not as e2e as in master, but this is because of diff modules and libraries we use in that one. I didnt wanted to start cherry-picking those. 

It does the job.

**Comment by sdodson** [comment]
@mjudeikis Do we intend to support this for 3.7?

**Comment by mjudeikis** [comment]
@sdodson Prometheus was tech previous in 3.7, but we have a customer who use it already. So it makes sense to backport "good stuff". My customer need this in particular. 
Grafana is unsupported and it uses @mrsiano image... So its not supported at all... But thing is that Prometheus without grafana is almost useless as metrics representation tool. 

No non of these things are supported (and grafana I'm not sure if we will support it in close future) but they are in big demand from the field...

I can add some "UNSUPORTED" warnings to playbooks if we chose so.

**Comment by sdodson** [comment]
Looks like playbooks/byo/openshift-grafana/README.md already mentions it's not supported. I'm fine with pulling these in pending @zgalor's review

File: roles/openshift_prometheus/tasks/install_prometheus.yaml [hunk 1]
```diff
@@ -1,124 +1,146 @@
...
     namespace: "{{ openshift_prometheus_namespace }}"
-    selector:
-      app: prometheus
     labels:
-      name: "{{ item.name }}"
-      #    TODO add annotations when supported
-      #    annotations:
-      #      service.alpha.openshift.io/serving-cert-secret-name: "{{item.name}}-tls"
+      name: prometheus
+    annotations:
```
Comment (zgalor) [inline roles/openshift_prometheus/tasks/install_prometheus.yaml]: I'm not sure oc_service support annotations in 3.7. is this tested?

Comment (zgalor) [inline roles/openshift_prometheus/tasks/install_prometheus.yaml]: I'm not sure oc_service supports annotations in 3.7.
Is this tested?

File: playbooks/byo/openshift-prometheus/README.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
+OpenShift Prometheus
```
Comment (zgalor) [inline playbooks/byo/openshift-prometheus/README.md]: I'm a bit confused. From where is this cherry-picked? There is no playbooks/byo/openshift-prometheus/ in master or 3.7.
In master and 3.7 the README.md is only inside the role and not around the playbook.

In 3.7 openshift-prometheus playbook is under byo/openshift-cluster.
I don't understand why the prometheus playbooks for 3.7 need to change in order to support grafana?

File: roles/openshift_prometheus/tasks/install_prometheus.yaml (hunk 1 above)
Comment (mjudeikis) [reply roles/openshift_prometheus/tasks/install_prometheus.yaml]: WIll confirm after next run. I think it does. But just to be sure

File: playbooks/byo/openshift-prometheus/README.md (hunk 2 above)
Comment (mjudeikis) [reply playbooks/byo/openshift-prometheus/README.md]: its not like it need to change. We have it as Tech-Prev, and our upstream example moved a little bit. My customer needs "new features" from upstream in their Tech-Preview deployment.
+ I'm adding entry points (playbooks) into 3.7 branches as despite the fact we had a role in the roles folder - it was kinda "DIY" when it came to run it...

Comment (zgalor) [reply playbooks/byo/openshift-prometheus/README.md]: 3.7 already has the entry points in the playbook:
https://github.com/openshift/openshift-ansible/blob/release-3.7/playbooks/common/openshift-cluster/openshift_prometheus.yml.
I didn't argue the roles additions. just the playbook.
What do changes in the playbook area add?

Comment (mjudeikis) [reply playbooks/byo/openshift-prometheus/README.md]: Playbooks entry in `byo` just makes things easy to understand for playbook consumer. As most of the playbooks follow same pattern. `byo` sets facts and calls `common`. 

I see I replicated whats in `common/openshift-cluster/openshift_prometheus.yml` and just created a separate dedicated folder (my bad). I think best way here would be to point `byo` to existing entry point in `openshift-cluster` so we would not duplicate them.

File: roles/openshift_prometheus/tasks/install_prometheus.yaml (hunk 1 above)
Comment (mjudeikis) [reply roles/openshift_prometheus/tasks/install_prometheus.yaml]: from python script I can see - yes:
form docs:
```
  annotations:
    description:
    - Annotations to apply to the object
    required: false
    default: None
    aliases: []
```

**Comment by mjudeikis** [comment]
@zgalor removed readme file (was in wrong place) and repointed directly o existing entry point just not to duplicate stuff.

**Comment by zgalor** [comment]
@mjudeikis Do you have a bz for this?
I think adding 3.9 features to 3.7 should go through @tkatarki

**Comment by mjudeikis** [comment]
Dont have any. At best it would be RFE. I can arrange customer to raise one
and it will get pushed via account managers. Its just my customer is going
live wit 3.7 and need these features.

**Comment by mjudeikis** [comment]
@zgalor I have few more customer asking for this. I dont think we adding 3.9 features, we just enabling capabilities for something we already shipped.

**Comment by zgalor** [comment]
@mjudeikis No problem. I still think we need an RFE.
Those small rebases/cherry-picks can break things (see https://bugzilla.redhat.com/show_bug.cgi?id=1540806), and 3.7 is already stable.

**Comment by mjudeikis** [comment]
@zgalor k, let me ask guys here to raise ticket so we have log

**Comment by mjudeikis** [comment]
@zgalor bz1555274 and bz1555272 
and this PR already needs rebasing

**Comment by zgalor** [comment]
@mjudeikis See @sdodson's replies in the BZs

**Pull request closed without merging**